clients/itwin/src/AuthorizationClient.ts

```diff
@@ -12,6 +12,6 @@ import { AccessToken } from "@itwin/core-bentley";
  * @beta
  */
 export interface AuthorizationClient {
-  /** Get the AccessToken of the currently authorized user. The token is refreshed if necessary and possible. */
-  getAccessToken(): Promise<AccessToken | undefined>;
+  /** Get the AccessToken of the currently authorized user, or blank string if no token is available. */
+  getAccessToken(): Promise<AccessToken>;
 }

```

Comment:
I appreciate that we will no longer need to check for `undefined` AccessToken state. We could also check whether `AccessToken` is valid at compile-time using the following types:
```ts
type ValidAccessToken = `${string} ${string}`;
type InvalidAccessToken = "";

const onlyValid: ValidAccessToken = ""; // Type Error
const onlyInvalid: InvalidAccessToken = "Bearer 123" // Type Error
```

Reply:
The implementation of getAccessToken should simply return the most-recently obtained value of the token, with no processing (I still vigorously maintain that it should **not** be async for that reason, @wgoehrig disagrees).

The validation you refer to should happen in the implementations of AuthorizationClient wherever the token is obtained/refreshed from an authority. If it can be determined to be invalid, the saved value should be "".

For mock tests, we allow bogus token values.

Reply:
Yeah, sadly there's no universal template for "non-empty string".